examples/other_datasets/financial_data.py

```diff
@@ -0,0 +1,184 @@
+        return all_epochs
+
+
+def slim(x, keep_diagonal=True):
+    # Vectorize covariance matrices by removing redundant information.
+    length = len(x) // 2
+    first = range(0, length)
+    last = range(len(x) - length, len(x))
+    down_cadrans = x[np.ix_(last, last)]
+    if keep_diagonal:
+        down_cadrans = [down_cadrans[i, j] for i in first for j in first if i <= j]
+    else:
+        down_cadrans = [down_cadrans[i, j] for i in first for j in first if i < j]
+    first_cadrans = np.reshape(x[np.ix_(last, first)], (1, len(x)))
+    ret = np.append(first_cadrans, down_cadrans)
+    return ret
+
+
+class SlimVector(TransformerMixin, BaseEstimator):
+    def __init__(self, keep_diagonal):
+        self.keep_diagonal = keep_diagonal
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        return np.array([slim(x, self.keep_diagonal) for x in X])
+
+
+class OptionalWhitening(TransformerMixin, BaseEstimator):
+    def __init__(self, process=True, n_components=4):
+        self.process = process
+        self.n_components = n_components
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        if not self.process:
+            return X
+        return Whitening(dim_red={"n_components": 4}).fit_transform(X)
+
+
+# Finally put together the transformers, and add at the end
+# the SVM classifier (classical)
+pipe = make_pipeline(
+    ToEpochs(n=10),
+    XdawnCovariances(nfilter=1),
+    OptionalWhitening(process=True, n_components=4),
+    SlimVector(keep_diagonal=True),
+    QuanticSVM(quantum=False),
+)
+
+# Optimize the pipeline.
+# Let's save some time and run the optimization with a classical SVM.
+gs = GridSearchCV(
+    pipe,
+    param_grid={
+        "toepochs__n": [10, 20],
+        "xdawncovariances__nfilter": [1, 2],
+        "optionalwhitening__process": [True, False],
+        "optionalwhitening__n_components": [2, 4],
+        "slimvector__keep_diagonal": [True, False],
+    },
+    scoring="balanced_accuracy",
+)
+
+##############################################################################
+# Run evaluation
+# ----------------
+#
+# Balance the data and run the evaluation on a quantum vs classical pipeline.
+
+# We only have about 200 frauds epochs over 30K entries.
+# Let's balance the problem using NearMiss.
+# Note: at this stage `features` also contains the `index` column.
+# So `NearMiss` we choose the closest 200 non-fraud epochs to the 200 fraud-epochs
+# based also on this `index` column. This should be improved for real use cases.
+X, y = NearMiss().fit_resample(features.to_numpy(), target.to_numpy())
+
+# Let's fit our GridSearchCV, to find the best hyper parameters:
+gs.fit(X, y)
+
+# This is the best score with the classical SVM.
+# /!\ Ideally, we should have different datasets for training and validation.
+# In a real scenario, we could use some data augmentation techniques, because
+# we have only a few samples.
```

Comment:
We have about 200 frauds only on the whole dataset. So I am not sure it will be able to generalize correctly if I take for example 100 fraud samples for training. I will try.